src/Services/QueuedJobService.php

```diff
@@ -159,18 +174,15 @@ public function __construct()
     /**
      * Adds a job to the queue to be started
      *
-     * Relevant data about the job will be persisted using a QueuedJobDescriptor
-     *
-     * @param QueuedJob $job
-     *          The job to start.
-     * @param $startAfter
-     *          The date (in Y-m-d H:i:s format) to start execution after
-     * @param int $userId
-     *          The ID of a user to execute the job as. Defaults to the current user
-     *
+     * @param QueuedJob $job The job to start.
+     * @param null|string $startAfter The date (in Y-m-d H:i:s format) to start execution after
+     * @param null|int $userId The ID of a user to execute the job as. Defaults to the current user
+     * @param null|int $queueName
+     * @param array $options options which are passed to the updateJobDescriptorBeforeQueued extension point
      * @return int
+     * @throws ValidationException
      */
-    public function queueJob(QueuedJob $job, $startAfter = null, $userId = null, $queueName = null)
+    public function queueJob(QueuedJob $job, $startAfter = null, $userId = null, $queueName = null, array $options = [])
```

Comment:
Adding this would be a breaking change, which would mean it'll take longer to get released. Is there a way you could implement this without changing the method signature?